sql/org_mozilla_firefox_derived/baseline_daily_v1/query.sql

```diff
@@ -0,0 +1,169 @@
+      baseline.ping_info.parsed_end_time,
+      udf.glean_timespan_seconds(baseline.metrics.timespan.glean_baseline_duration) AS duration,
+      baseline.client_info.android_sdk_version,
+      baseline.client_info.app_build,
+      baseline.client_info.app_channel,
+      baseline.client_info.app_display_version,
+      baseline.client_info.architecture,
+      baseline.client_info.device_manufacturer,
+      baseline.client_info.device_model,
+      baseline.client_info.telemetry_sdk_build,
+      baseline.client_info.locale,
+      baseline.metadata.geo.city,
+      baseline.metadata.geo.country,
+      baseline.metrics.string.glean_baseline_locale,
+      baseline.normalized_os,
+      baseline.normalized_os_version,
+  )
+);
+
+WITH unioned AS (
+  SELECT
+    extract_fields(baseline).*,
+    'release' AS normalized_channel
+  FROM
+    org_mozilla_firefox.baseline AS baseline
+  UNION ALL
+  SELECT
+    extract_fields(baseline).*,
+    'beta' AS normalized_channel
+  FROM
+    org_mozilla_firefox_beta.baseline AS baseline
+  UNION ALL
+  SELECT
+    extract_fields(baseline).*,
+    'nightly' AS normalized_channel
+  FROM
+    org_mozilla_fenix.baseline AS baseline
+  -- These final two apps should be retired in February.
+  UNION ALL
+  SELECT
+    extract_fields(baseline).*,
+    'aurora nightly' AS normalized_channel
+  FROM
+    org_mozilla_fennec_aurora.baseline AS baseline
+  UNION ALL
+  SELECT
+    extract_fields(baseline).*,
+    'preview nightly' AS normalized_channel
+  FROM
+    org_mozilla_fenix_nightly.baseline AS baseline
+),
+--
+with_dates AS (
+  SELECT
+    *,
+    -- For explanation of session start time calculation, see Glean docs:
+    -- https://mozilla.github.io/glean/book/user/pings/baseline.html#contents
+    DATE(TIMESTAMP_SUB(parsed_end_time, INTERVAL duration SECOND)) AS session_start_date,
+    DATE(parsed_end_time) AS session_end_date,
+  FROM
+    unioned
+),
+--
+base AS (
+  SELECT
+    *,
+    DATE_DIFF(submission_date, session_start_date, DAY) AS session_start_date_offset,
+    DATE_DIFF(submission_date, session_end_date, DAY) AS session_end_date_offset,
+  FROM
+    with_dates
+),
+--
+windowed AS (
+  SELECT
+    submission_date,
+    client_id,
+    sample_id,
+    ROW_NUMBER() OVER w1_unframed AS _n,
+    --
+    -- Take the earliest first_run_date if ambiguous.
+    MIN(first_run_date) OVER w1 AS first_run_date,
+    --
+    -- Sums over distinct baseline pings.
+    SUM(IF(duration BETWEEN 0 AND 100000, duration, 0)) OVER w1 AS durations,
+    --
+    -- Bit patterns capturing activity dates relative to the submission date.
+    BIT_OR(
+      1 << IF(session_start_date_offset BETWEEN 0 AND 27, session_start_date_offset, NULL)
+    ) OVER w1 AS days_seen_session_start_bits,
+    BIT_OR(
+      1 << IF(session_end_date_offset BETWEEN 0 AND 27, session_end_date_offset, NULL)
+    ) OVER w1 AS days_seen_session_end_bits,
```

Comment:
Went ahead and created bit patterns for session start and session end dates instead of date arrays. We can revisit this decision in the future.